src/Bedrock.Framework/Protocols/ProtocolWriter.cs

```diff
@@ -72,9 +72,13 @@ public async ValueTask WriteManyAsync<TWriteMessage>(IMessageWriter<TWriteMessag
                     return;
                 }
 
-                foreach (var protocolMessage in protocolMessages)
+                var enumerator = protocolMessages.GetEnumerator();
+                while(enumerator.MoveNext())
```

Comment:
Why did you switch from `foreach` to the `while` loop?

Reply:
Yes it should be changed back and you can use the same local variable trick to pass by ref (though I'm not sure about that in general).

Reply:
Didn't even think about the local variable reassign trick.